plugins/module_utils/template.py

```diff
@@ -291,3 +300,140 @@ def get_l3out_interface_routing_policy_object(self, uuid=None, name=None, fail_m
                 fail_module,
             )
         return existing_l3out_interface_routing_policy  # Query all objects
+
+    def get_tenant_policy_uuid(self, tenant_template, policy_name, policy_type):
+        """
+        Get the UUID of a Tenant Policy by name.
+        :param tenant_template: The tenant object -> Dict
+        :param policy_name: Name of the policy -> Str
+        :param policy_type: The type of the policy specified in the API response -> Str
+        :return: UUID of the tenant policy -> Str
+        """
+        existing_policies = tenant_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get(policy_type, [])
+        match = self.get_object_by_key_value_pairs(policy_type, existing_policies, [KVPair("name", policy_name)], fail_module=True)
+        return match.details.get("uuid")
+
+    @staticmethod
```

Comment:
Why is it set to `staticmethod`? Same question for other `staticmethod`.

Also we can access the `mso` directly from the `MSOTemplate` using `self.mso`.